### PR TITLE
Callouts in DCR - PR 4/5 - Add new CalloutBlockElement

### PR DESCRIPTION
**Pull request by tkgnm**
# Callouts in DCR

This PR is part of a larger body of work to create a new Callout component in DCR. We are separating this PR into chunks so that it is easier to communicate the intent of each PR and to make it easier for DCR to review. After each PR is merged, we can merge it into the next branch which will incrementally reduce the size of each subsequent PR. 

Important note for DCR reviewers: Please review these PR's in order below! 

PR 1 https://github.com/guardian/dotcom-rendering/pull/6741
PR 2 https://github.com/guardian/dotcom-rendering/pull/6819
PR 3 https://github.com/guardian/dotcom-rendering/pull/6833
PR 4 https://github.com/guardian/dotcom-rendering/pull/6806
PR 5 https://github.com/guardian/dotcom-rendering/pull/6836

## Screenshots

| Before      | After      |
|-------------|------------|
| ![before][] | ![after][] |

[before]: https://example.com/before.png
[after]: https://example.com/after.png

### CODE REVIEW

File: dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import { css } from '@emotion/react';
+import { palette, textSans, space } from '@guardian/source-foundations';
+import { SvgClock } from '@guardian/source-react-components';
+
+type Props = {
+	until?: number | undefined;
+};
+
+// work
```
Comment (sophie-macmillan) [inline dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx:9]: nit: remove comment

File: dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import { css } from '@emotion/react';
+import { palette, textSans, space } from '@guardian/source-foundations';
+import { SvgClock } from '@guardian/source-react-components';
+
+type Props = {
+	until?: number | undefined;
```
Comment (sophie-macmillan) [inline dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx:6]: Think you don't need the `|undefined` here, as its an optional `?:`

File: dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	const deadlineText = getDeadlineText(now, untilDate);
+	if (!deadlineText) return null;
+
+	return (
+		<span css={deadlineStyles}>
+			<SvgClock size="xsmall" />
+			{deadlineText}
+		</span>
+	);
+};
```
Comment (sophie-macmillan) [inline dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx:56]: could we pull this into common rendering for both dcr and ar as they are the same, or are we moving away from common rendering completely?

File: dotcom-rendering/src/web/components/Callout/CalloutShare.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+const shareCalloutLinkStyles = (format: ArticleFormat) =>
+	css`
+		color: ${decidePalette(format).text.richLink};
+		border-bottom: 1px solid ${decidePalette(format).text.richLink};
+		text-decoration: none;
+		font-weight: normal;
+	`;
+
+const placeholderCircle = css`
```
Comment (sophie-macmillan) [inline dotcom-rendering/src/web/components/Callout/CalloutShare.tsx:24]: Can we remove this placeholder now we have the icon in source?

File: dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx (hunk 3 above)
Comment (tkgnm) [reply dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx:56]: I think the latter - probably best to just put this in source. @abeddow91 may know more.

File: dotcom-rendering/src/web/components/Callout/CalloutShare.tsx (hunk 4 above)
Comment (tkgnm) [reply dotcom-rendering/src/web/components/Callout/CalloutShare.tsx:24]: ah yes thanks - I think this stying is still needed I just need to rename it

File: dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx (hunk 2 above)
Comment (tkgnm) [reply dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx:6]: Thanks - removed

File: dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx (hunk 1 above)
Comment (tkgnm) [reply dotcom-rendering/src/web/components/Callout/CalloutDeadline.tsx:9]: 👍

**Pull request closed without merging**